shared/instance.go

```diff
@@ -114,19 +114,27 @@ var InstanceConfigKeysAny = map[string]func(value string) error{
 		}
 
 		if strings.HasSuffix(value, "%") {
-			_, err := strconv.ParseInt(strings.TrimSuffix(value, "%"), 10, 64)
+			num, err := strconv.ParseInt(strings.TrimSuffix(value, "%"), 10, 64)
 			if err != nil {
 				return err
 			}
 
+			if num == 0 {
+				return fmt.Errorf("Memory limit can't be 0")
```

Comment:
This should probably state `0%` then.

Reply:
True indeed.

Reply:
Needs to be `0%%` actually to avoid issues.

Reply:
Yeah I always forget that :)

Reply:
This actually doesn't seem to work with `fmt.Errorf`, I know @tomponline mentioned we're avoiding using `errors`. Would this be an exception?

Reply:
Works here https://play.golang.org/p/QE2SsE0jZKk

What version of go are you using?

Reply:
Works if you `fmt.Println` the error afterwards, but I don't think that's how we print errors normally.

Reply:
Oops, accidentally pushed an empty commit :)

Reply:
Ah you can use the stdlib errors.New() then

Reply:
Should be good to go then, thanks for the input :)